src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -81,21 +84,32 @@ public static LoginData performAutoLogin(StartupParams params) {
         loginData.setCredentials(username, password);
 
         try {
-            System.out.println("Auto logging in (1/2)");
-            usernameLogin(loginData);
+            if(isSidAutoLogin) loginData.setSid(sid, server + ".darkorbit.com");
+            else {
+                System.out.println("Auto logging in (1/2)");
+                usernameLogin(loginData);
+            }
             System.out.println("Loading spacemap (2/2)");
             findPreloader(loginData);
         } catch (IOException e) {
             System.err.println("IOException trying to perform auto login, servers may be down");
             e.printStackTrace();
         } catch (WrongCredentialsException e) {
+            if(isSidAutoLogin) {
+                System.err.println("Wrong SID");
+                params.clearAutoLoginProp(StartupParams.PropertyKey.SID);
+                return performAutoLogin(params);
+            }
             System.err.println("Wrong credentials, check your username and password");
         }
 
         if (loginData.getPreloaderUrl() == null || loginData.getParams() == null) {
             System.err.println("Could not find preloader url or parameters, exiting bot.");
             System.exit(-1);
         }
+
+        if(!Strings.isEmpty(server) && !isSidAutoLogin) params.updateSidAndServer(loginData.getSid(), loginData.getUrl().split("\\.")[0]);
```

Comment:
I think this should be a prop in its own, like `UPDATE_SID=true` in the file, you're assuming here that just because we have a server we want to print the sid which seems a bit odd
```suggestion
        if (!isSidLogin && Boolean.parseBoolean(params.getAutoLoginValue(StartupParams.PropertyKey.UPDATE_SID)))
            params.updateSidAndServer(loginData.getSid(), loginData.getUrl().split("\\.")[0]);
```
additionally, you could probably do this right after calling `usernameLogin` as if that did not throw an exception, you there know *for sure* the sid will be fresh

Reply:
I think flag is not necessary because if user want use only sid than properties should not contain login/password.
I not see scenario when user want use sid and login and not want store actual sid iin case present is wrong

Reply:
You may not have SID atm, but setup user & pass with the intention that they will save, and get re-used in future uses

Reply:
Now role of this flag keep the `server` field.
BUT
I think may join `server` and `sid` to single line like prop file will contain `sid=eu1:yyyySiDIIIIiiiiiiiiiiiiiiiiiiiiiiiiiiii`, and existing of this field will be flag for store new sid. I mean if user not configure sid in props than props file never update.
Because now it look like that autologin by user/password will switch to server that declared in `server` field.

If sid wrong and we gets new than why we should keep expired sid in props.

Reply:
I think joining server & sid on the same prop is a bad idea, generally you want data to be atomic
it also makes it more intuitive, server and sid are basically never combined, and shouldn't be here either

how would it be if we had `user=username:pass`, it'd look pretty weird (ignoring that : could be part of user or pass so it'd be unreliable, of course)

I think they should be kept separate